src/ingest_validation_tools/directory_validator.py

```diff
@@ -3,6 +3,38 @@
 from fnmatch import fnmatch
 from importlib import import_module
 
+def default_factory(base_path, assay_type):
+    """
+    default validator factory
+    """
+    return DefaultDirectoryContentValidator(base_path, assay_type)
+
+
+DIRECTORY_CONTENT_VALIDATOR_FACTORY = default_factory
+
+
+class DirectoryContentValidator(object):
+    def __init__(self, base_path, assay_type):
+        self.base_path = base_path
+        self.assay_type = assay_type
+
+    def collect_errors(self):
+        """
+        returns string or None.  Multiple errors are to be reported
```

Comment:
Is there are reason not to return an array? For that matter, I'm not sure restricting to strings is necessary, if dicts could be useful.

Reply:
Sorry, I thought string or none was your intention with the design.  I'd prefer returning an array, as long as we specify the types of the array elements.